taskcluster/async/asyncclient.py

```diff
@@ -144,7 +143,7 @@ class AsyncBaseClient(BaseClient):
                 if response.status == 204:
                     return None
 
-            except aiohttp.HttpProcessingError as rerr:
+            except aiohttp.ClientError as rerr:
```

Comment:
Thinking about this a little more, since the default in aiohttp is to follow redirects, and `raise_for_status()` only raises for http status code >= 400, and that not all 300s are automatically follow-able redirects, and none (I think) contain a valid response body, we should probably check the status code and throw if <200 or >= 300 and not use `raise_for_status()`.

The requests library also has this issue.  Would you be able to update the sync client to match this behaviour?  The aiohttp docs seem to be better than the requests docs, since the latter requires you to read the code to spot that only >=400 is throwing.